src/Util/ClassSourceManipulator.php

```diff
@@ -178,6 +180,40 @@ public function addInterface(string $interfaceName)
         $this->updateSourceCodeFromNewStmts();
     }
 
+    public function addTrait($classTraitName)
+    {
+        $this->addUseStatementIfNecessary($classTraitName);
+
+        $classNode = $this->getClassNode();
+        $name = Str::getShortClassName($classTraitName);
+
+        if (empty($classNode->stmts)) {
+            $classNode->stmts[0] = new Node\Name('use '.$name.';');
+        } else {
+            $traitNode = $this->findFirstNode(function ($node) {
+                return $node instanceof Node\Stmt\Class_
+                    && !empty($node->stmts[0])
+                    && $node->stmts[0] instanceof Node\Stmt\TraitUse;
+            });
+
+            if ($traitNode) {
+                foreach ($classNode->stmts as $value) {
+                    if (!$value instanceof Node\Stmt\TraitUse) {
+                        break;
+                    }
+                    // if trait already exists
+                    if (strstr($value->traits[0]->parts[0], $name)) {
```

Comment:
We typically have used something like `$value->traits[0]->toString()`. Basically, when working with a `PhpParser\Node\Name` instance (which is what `$value->traits[0]` is here) we try to call a method on it, instead of working with the `->parts` array.

Reply:
Done.